lib/plugins/aws/package/compile/events/httpApi/index.js

```diff
@@ -135,6 +135,14 @@ class HttpApiEvents {
       },
     };
 
+    if (this.config.throttle.burstLimit) {
```

Comment:
Shouldn't we set a default value here? There is an issue when changes removed from configuration are not reflected on (re)deployment.  See https://github.com/DianaIonita/serverless-api-gateway-throttling/issues/5.

Reply:
I made it -1 by default.

Reply:
Sorry, I didn't see that comment, when doing previous review. 

> There is an issue when changes removed from configuration are not reflected on (re)deployment.

If it's the case, then I agree, let's always set this property.

Still if it's `-1` that's default, then we should also allow it by schema (currently it's `0` minimum there).

Unfortunately I'm unable to find any info in AWS docs, on what's the range of allowed values

Reply:
@medikoo, the workaround with `-1` as the default value was found in AWS Terraform provider and I probably saw it somewhere else, too. Unfortunately it's not possible to remove throttling settings for method at the moment and that's the only viable option for now. Setting `-1`  ensures that custom settings are properly removed from method-specific settings - this has been proven with Terraform and the sls plugin mentioned in my initial comment.

Reply:
> this has been proven with Terraform and the sls plugin mentioned in my initial comment.

Thanks, still can you validate this implementation, but doing test deployments with this branch?

Terrraform I think doesn't use CloudFormation, not sure about plugin. We really need to confirm that those values bring desired effect as set in CloudFormation template as we do with this PR

Reply:
@medikoo I just realized that the case with the default as `-1` applies to the method-specific settings, so that's not the scope of this PR (yet?).
The issue was not CloudFormation here, but the AWS API's that doesn't allow removing throttling settings from the method - it'd be simply ignored. It can be even observed in AWS Console - `-1` is set for `burstLimit` and `rateLimit` with `replace` rather than `delete` as we might assume.

Reply:
Ok, thanks for clarification. Here we should focus purely on how deployments work through CloudFormation template.

So let's get back to the beginning: Will removing the property from CF template, reset the actual property value to the default? If it's the case, then we definitely should not set them if user do not configure them in service config

Reply:
As far as I can see previously set throttling settings on stage are not reset to the default (account level throttling) when not present in the CloudFormation template.

Reply:
Ok, and what value as set to CloudFormation template, ensures that with CloudFormation deployment value is reset to default (so not throttling is set) ?

Reply:
@medikoo it looks like throttling setting for the default route can be only "reset" by using account level limits. I'm not sure if that's something that CloudFormation. `-1` also doesn't work in this case (only values from `0` up to account level max are accepted) and defaulting to `0` results with throttling every request. So far I couldn't find anything to properly reset this setting without recreating the whole stage 😞.

Reply:
I will ask the AWS Support on how to achieve this.

Reply:
> As per my testing, I can confirm that once configured, throttling burst limits can't be disabled neither from CloudFormation nor API Gateway. [...] I have engaged the API Gateway Service Team for more clarity.

Waiting for a reply from the API Gateway Service Team :)